### DISCUSSION TITLE
Export equivalent circuit for non nominal design does not seems to work

### DISCUSSION THREAD

**Discussion by npanti** [ansys/pyaedt#2599] [Q&A] [answered]
Using the example provided on the PyAnsys website to simulate strip lines with Q2D, I cannot export the equivalent circuit for design variations that are not nominal.

The example is on this page
https://aedt.docs.pyansys.com/release/0.6/examples/05-Q3D/Q2D_Example_Stripline.html#sphx-glr-examples-05-q3d-q2d-example-stripline-py

I am just adding the following command 

q.export_equivalent_circuit(file_name='test.ckt', 
              sweep='sweep1',
              variations=['sig_bot_w:75um','sig_gap:100um'],
              matrix_name='diff_pair')

But I get the error: "pyaedt ERROR: Provided variation value doesn't exist."

How should I define the variations parameters? It is not clear to me.

Looking at the source code, it seems that you can only export the nominal variation.
https://github.com/pyansys/pyaedt/blob/2475df2ac233a2d657c2af588240e8b3f9807aba/pyaedt/q3d.py#L935

Thanks,

Nicolas

**Comment by Samuelopez-ansys** [comment, marked as answer]
@npanti You can use the following code to export any variation:

```
variation_number = 3

variation_list = q.list_of_variations()
variation = variation_list[variation_number ]
input_string = variation.replace("'", "")
pairs = input_string.split()
formatted_list = [f"{pair.split('=')[0]}:{pair.split('=')[1]}" for pair in pairs]

q.export_equivalent_circuit(
    file_name='test2.ckt',
    sweep='sweep1',
    variations=formatted_list,
    matrix_name='diff_pair')
```
